File: Leetcode/Java/Q1512_Number_of_Good_Pairs.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
        // Array to count occurrences of each number (assuming numbers range from 0 to 100)
        int[] count = new int[101];
        // Variable to store the total number of good pairs
        int ans = 0;
        
        // Iterate through each number in the input array
        for (int x : nums) {
            // Increment the number of good pairs by the current count of 'x'
            // The count[x] represents the number of times 'x' has appeared before
            // Therefore, each of those appearances forms a good pair with the current 'x'
            ans += count[x]++;
        }
        
        // Return the total number of good pairs
        return ans;
    }
}

File: Leetcode/Python/Q1512_Number_of_Good_Pairs.py
class Solution(object):
    def numIdenticalPairs(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        # Create a list to count the occurrences of each number from 0 to 100
        count = []
        for i in range(101):
            count.append(0)

        ans = 0
        # Iterate through the list of numbers
        for i in range(len(nums)):
            # Add the current count of the number to the answer
            ans += count[nums[i]]
            # Increment the count of the current number
            count[nums[i]] += 1
        return ans
